src/Temporalio/Worker/TemporalWorkerOptions.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Temporalio.Worker
+{
+    /// <summary>
+    /// Options for a <see cref="TemporalWorker" />. <see cref="TaskQueue" /> and at least one
+    /// workflow or activity are required.
+    /// </summary>
+    public class TemporalWorkerOptions : ICloneable
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TemporalWorkerOptions"/> class.
+        /// </summary>
+        public TemporalWorkerOptions()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TemporalWorkerOptions"/> class.
+        /// </summary>
+        /// <param name="taskQueue">Task queue for the worker.</param>
+        public TemporalWorkerOptions(string taskQueue)
+        {
+            TaskQueue = taskQueue;
+        }
+
+        /// <summary>
+        /// Gets or sets the task queue for the worker.
+        /// </summary>
+        public string? TaskQueue { get; set; }
```

Comment:
It has to be, as is every required option without a default. This is the same for host port for client connection even though I will fail at runtime if not set. It is the .NET options pattern approach to allow all of these to be late bound instead of required on init (and using placeholders like an empty string is unclear in that it makes one think it has a default and that default is acceptable).

Reply:
Only via docs and strong hints by having it in a constructor if I'm following options pattern. I really wish I could use records and required init properties. The .NET options guidelines don't say much on this, see https://learn.microsoft.com/en-us/dotnet/core/extensions/options-library-authors. But the ASP.NET guidelines at https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options do state:

> Must be non-abstract with a public parameterless constructor.
> All public read-write properties of the type are bound.

Mostly due to JSON deserialization that needs to be able to construct the object before deserializing.